File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    answer = 0
    dp = [0 for _ in range(2010)]
    dp[1] = 1
    dp[2] = 2
    dp[3] = 3
    dp[4] = 5
    
    if n > 4:
        for i in range(4,n+1):
            dp[i] = dp[i-1]%1234567 + dp[i-2]%1234567

    return dp[n]%1234567

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
  if (n === 1 || n === 2) return n;
  const dp = Array(n).fill(0);
  const mod = 1234567;
  dp[0] = 1;
  dp[1] = 2;
  for (let i = 2; i < n; i++) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % mod;
  }
  return dp[n - 1];
}